sklearn/tree/_tree.pyx

```diff
@@ -428,39 +605,46 @@ cdef class Entropy(ClassificationCriterion):
         cdef double entropy_right = 0.0
         cdef double total_left = 0.0
         cdef double total_right = 0.0
-        cdef double tmp
+        cdef double pmk
         cdef SIZE_t k
         cdef SIZE_t c
 
+        # For each response
         for k in range(n_outputs):
+            # Initialize the left and right entropies to be 0
             entropy_left = 0.0
             entropy_right = 0.0
 
+            # For each unique class in this response
             for c in range(n_classes[k]):
-                tmp = label_count_left[c]
-                if tmp > 0.0:
-                    tmp /= weighted_n_left
-                    entropy_left -= tmp * log(tmp)
-
-                tmp = label_count_right[c]
-                if tmp > 0.0:
-                    tmp /= weighted_n_right
-                    entropy_right -= tmp * log(tmp)
-
+                # Calculate this classes entropy on the left node
+                pmk = label_count_left[c]
+                if pmk > 0.0:
+                    pmk /= weighted_n_left
+                    entropy_left -= pmk * log(pmk)
+
+                # Calculate this classes entropy on the right node
+                pmk = label_count_right[c]
+                if pmk > 0.0:
+                    pmk /= weighted_n_right
+                    entropy_right -= pmk * log(pmk)
+
+            # Add local entropy to the total entropy on the appropriate side
             total_left += entropy_left
             total_right += entropy_right
             label_count_left += label_count_stride
             label_count_right += label_count_stride
 
+        # Save the average entropy over all responses
         impurity_left[0] = total_left / n_outputs
         impurity_right[0] = total_right / n_outputs
 
 
 cdef class Gini(ClassificationCriterion):
-    """Gini Index impurity criteria.
-
-    Let the target be a classification outcome taking values in 0, 1, ..., K-1.
-    If node m represents a region Rm with Nm observations, then let
+    """
+    A class representing the Gini Index impurity criteria. This handles
```

Comment:
Not of fan of "A class representing [something]" instead of "[Something]". But this is only my opinion.

(We dont document methods as "A method returning [...]", dont see why we should do that for classes)